proposals/0759-react-native-frameworks.md

```diff
@@ -0,0 +1,454 @@
+
+### Template
+
+The React Native template is located inside the `/template` folder of the NPM package ([visible here](https://github.com/facebook/react-native/tree/main/packages/react-native/template)) and offers a mechanism to create a simple Android & iOS app which runs React Native.
+
+A new project can be scaffolded from the template by using the community CLI, and the `npx react-native init` command (which delegates to` @react-native-community/cli init `behind the scenes).
+
+Practically, the template represents a **minimal integration** of the React Native **APIs**. Those APIs are partially described in the “[Integrating with Existing Apps](https://reactnative.dev/docs/integration-with-existing-apps)”. In an ideal world, if you were to:
+
+1. Create a new Android/iOS project on Android Studio/Xcode
+2. Run through [all the steps in this page](https://reactnative.dev/docs/integration-with-existing-apps).
+
+you will effectively **recreate the template**.
+
+The template gets more controversial as it’s **not a Framework** but it offers some of the capabilities that are specific to Frameworks. For example it offers an auto-linking implementation, and a mechanism to help with upgrades (the upgrade helper). It lacks other crucial features of frameworks, such as routing, navigation or support for publishing apps.
+
+To fully understand the role of the template, it’s crucial we understand what are the use cases of the template as of today.
+
+The React Native template is used for:
+
+1. **Internal** use cases (Meta engineers/Contributors)
+    1. The React Native CI is using it to test that a new project can be successfully created on every PR/Commit ([example job definition here](https://github.com/facebook/react-native/blob/28dfdb22ebaff7055fd8bb4b8847e018181f849d/.circleci/config.yml#L769))
+    2. The Release Crew is using it to test that a new app can be created, built and executed correctly **before releasing** a new version of React Native ([docs on this process are here](https://reactnative.dev/contributing/release-testing))
+    3. Issues and bug reports require a **reproducer**, being either an Expo Snack or an instance of a template project ([see here for the reproducer](https://github.com/react-native-community/reproducer-react-native)).
+2. **External** use cases
+    1. React Native developers that are **not using a Framework** because they haven’t migrated to it yet
+    2. React Native developers that are **not using a Framework** such as Expo because it doesn’t satisfy their requirements
```

Comment:
Is it worth to directly reference the need for using out-of-tree platforms not supported by Expo (because some are supported)?

Reply:
Yup I've added a sentence about it here b5f463f